src/sif/truckcost.cc

```diff
@@ -434,18 +434,20 @@ bool TruckCost::AllowedReverse(const baldr::DirectedEdge* edge,
 Cost TruckCost::EdgeCost(const baldr::DirectedEdge* edge,
                          const baldr::GraphTile* tile,
                          const uint32_t seconds) const {
-  auto speed = tile->GetSpeed(edge, flow_mask_, seconds);
+  auto edge_speed = tile->GetSpeed(edge, flow_mask_, seconds);
 
-  // Use the lower or truck speed (ir present) and speed
-  uint32_t s = (edge->truck_speed() > 0) ? std::min(edge->truck_speed(), speed) : speed;
-  assert(s < speedfactor_.size());
+  // Use the lower of truck speed, top_speed (if present) and speed
+  uint32_t s = (edge->truck_speed() > 0) ? std::min({edge->truck_speed(), edge_speed, top_speed_})
+                                         : std::min(edge_speed, top_speed_);
```

Comment:
since we got truck speed above we can simplify

```suggestion
  uint32_t s = std::min(edge_speed, top_speed_);
```